test/sqllogictest/github-11111.slt

```diff
@@ -0,0 +1,28 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Regression test for https://github.com/MaterializeInc/materialize/issues/11111
+# See https://github.com/MaterializeInc/materialize/issues/11111
+
+# Verify that parsing of ALTER INDEX works as expected
+
+statement ok
+CREATE TABLE t1 (f1 INTEGER, f2 INTEGER);
+
+statement ok
+CREATE INDEX i1 ON t1 (f2);
+
+statement error Expected one of RESET or SET or RENAME, found semicolon
+ALTER INDEX i1;
```

Comment:
Can you instead move these tests into `src/sql-parser/tests/testdata/ddl`? That's where the parsing tests live; sqllogictest is moreso for testing the "runtime" behavior of SQL statements